src/transport/SecureSessionMgr.cpp

```diff
@@ -160,6 +161,12 @@ CHIP_ERROR SecureSessionMgr::SendMessage(SecureSessionHandle session, PayloadHea
     VerifyOrExit(admin != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
     localNodeId = admin->GetNodeId();
 
+    if (payloadHeader.HasMessageType(Protocols::SecureChannel::MsgType::MsgCounterSyncReq) ||
+        payloadHeader.HasMessageType(Protocols::SecureChannel::MsgType::MsgCounterSyncRsp))
+    {
+        packetHeader.SetSecureSessionControlMsg(true);
+    }
+
```

Comment:
Can we move these logic into `MessageCounterSyncMgr::SendMsgCounterSyncReq`

Reply:
That was the original plan, but we need to add packetHeader argument to current SecureSessionMgr API. Per Boris's suggestion, we can hard-code the C flag for "MsgCounterSyncReq" and "MsgCounterSyncRsp" explicitly for now since only these two messages are using this flag. https://github.com/project-chip/connectedhomeip/pull/4771